x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -477,126 +475,42 @@ export class ObservabilityAIAssistantClient {
       simulateFunctionCalling?: boolean;
       tracer: LangTracer;
     }
-  ): Observable<ChatCompletionChunkEvent | TokenCountEvent> => {
-    return defer(() =>
-      from(
-        withAssistantSpan('get_connector', () =>
-          this.dependencies.actionsClient.get({ id: connectorId, throwIfSystemAction: true })
-        )
-      )
-    ).pipe(
-      switchMap((connector) => {
-        this.dependencies.logger.debug(`Creating "${connector.actionTypeId}" adapter`);
-
-        let adapter: LlmApiAdapter;
-
-        switch (connector.actionTypeId) {
-          case ObservabilityAIAssistantConnectorType.OpenAI:
-            adapter = createOpenAiAdapter({
-              messages,
-              functions,
-              functionCall,
-              logger: this.dependencies.logger,
-              simulateFunctionCalling,
-            });
-            break;
-
-          case ObservabilityAIAssistantConnectorType.Bedrock:
-            adapter = createBedrockClaudeAdapter({
-              messages,
-              functions,
-              functionCall,
-              logger: this.dependencies.logger,
-            });
-            break;
-
-          case ObservabilityAIAssistantConnectorType.Gemini:
-            adapter = createGeminiAdapter({
-              messages,
-              functions,
-              functionCall,
-              logger: this.dependencies.logger,
-            });
-            break;
-
-          default:
-            throw new Error(`Connector type is not supported: ${connector.actionTypeId}`);
-        }
-
-        const subAction = adapter.getSubAction();
-
-        if (this.dependencies.logger.isLevelEnabled('trace')) {
-          this.dependencies.logger.trace(JSON.stringify(subAction.subActionParams, null, 2));
-        }
-
-        return from(
-          withAssistantSpan('get_execute_result', () =>
-            this.dependencies.actionsClient.execute({
-              actionId: connectorId,
-              params: subAction,
-            })
-          )
-        ).pipe(
-          switchMap((executeResult) => {
-            if (executeResult.status === 'error' && executeResult?.serviceMessage) {
-              const tokenLimitRegex =
-                /This model's maximum context length is (\d+) tokens\. However, your messages resulted in (\d+) tokens/g;
-              const tokenLimitRegexResult = tokenLimitRegex.exec(executeResult.serviceMessage);
-
-              if (tokenLimitRegexResult) {
-                const [, tokenLimit, tokenCount] = tokenLimitRegexResult;
-                throw createTokenLimitReachedError(
-                  parseInt(tokenLimit, 10),
-                  parseInt(tokenCount, 10)
-                );
-              }
-            }
-
-            if (executeResult.status === 'error') {
-              throw createInternalServerError(
-                `${executeResult?.message} - ${executeResult?.serviceMessage}`
-              );
+  ): Observable<ChatCompletionEvent> => {
```

Comment:
Ideally we encapsulate the event type from the inference client here, and convert it into `ChatCompletionChunkEvent`. We should eventually completely drop `ChatCompletionChunkEvent` in favor of the inference plugin's types, but that means that we need to make changes in many places, and there's some stuff that isn't supported yet by the Inference plugin's types. Until that happens, I think we should limit the blast radius and stick to our own types where we can - which should be everywhere except for the implementation of this function, I think.

Reply:
agree, I will use ChatEvent to maintain compatibility with existing types (`ChatCompletionChunkEvent`, and `TokenCountEvent`) and adding `InferenceChatCompletionChunkEvent`

Reply:
To clarify, what I mean is that we only have `ChatCompletionChunkEvent` (which is exclusive to the Obs AI Assistant). Any event from the Inference client is ideally not exposed via the `chat` method. So the signature of the `chat` method should stay the same, and existing types as well. Does that make sense?

Reply:
Yes, that makes sense. I’ll keep the `ChatCompletionChunkEvent` as the main event type for the `ObservabilityAIAssistantClient.chat` method to ensure we don’t expose any inference-client-specific types directly. I’ll handle the conversion internally within the implementation, encapsulating the inference client events as needed without altering the existing method signature or exposing the `InferenceChatCompletionChunkEvent`. 
- Solution:
Within the pipe, we’ll add a mapping step to convert any incoming `InferenceChatCompletionChunkEvent` to a `ChatCompletionChunkEvent` type. should I do it for all the event emitted from `inferenceClient.complete` method?
for these 3 inference client events:
```ts
export type ChatCompletionEvent<TToolOptions extends ToolOptions = ToolOptions> =
  | ChatCompletionChunkEvent
  | ChatCompletionTokenCountEvent
  | ChatCompletionMessageEvent<TToolOptions>;
```